db/db_impl/db_impl_readonly.cc

```diff
@@ -130,10 +130,20 @@ Status DBImplReadOnly::NewIterators(
 
 Status DB::OpenForReadOnly(const Options& options, const std::string& dbname,
                            DB** dbptr, bool /*error_if_log_file_exist*/) {
+  // Attempt to read "CURRENT" file
```

Comment:
This looks good, but if ldb is called with --column-family=foo, then ldb will directly call the other overload of OpenForReadOnly. So we must check for that overload also.

But preferably we would not check twice on calling this overload, because we do not want unnecessary I/Os on DB open.

And since this affects all users of DB::OpenForReadOnly, I think we need to add to HISTORY.md as a "Public API Change". (This is also kind of a bug fix, but I think it's better to call it a "change".) Suggested wording:
* DB::OpenForReadOnly no longer creates files or directories if the named DB does not exist, unless create_if_missing is set to true.

Speaking of which, this check should not be made if create_if_missing is set to true, as weird as that option is for OpenForReadOnly

Reply:
I didn't know the other overload may be called. Thanks for bringing this up!
I'll also try to figure out if there is a way to avoid the checking twice.
(will change HISTORY.md accordingly in the next update.)